ignite/handlers/early_stopping.py

```diff
@@ -54,37 +64,50 @@ def __init__(
         if not callable(score_function):
             raise TypeError("Argument score_function should be a function.")
 
+        if not callable(pass_function):
+            raise TypeError("Argument pass_function should be a function.")
+
+        if not callable(stop_function):
+            raise TypeError("Argument stop_function should be a function.")
+
+        if not isinstance(trainer, Engine):
+            raise TypeError("Argument trainer should be an instance of Engine.")
+
         if patience < 1:
             raise ValueError("Argument patience should be positive integer.")
 
         if min_delta < 0.0:
             raise ValueError("Argument min_delta should not be a negative number.")
 
-        if not isinstance(trainer, Engine):
-            raise TypeError("Argument trainer should be an instance of Engine.")
-
-        self.score_function = score_function
         self.patience = patience
-        self.min_delta = min_delta
-        self.cumulative_delta = cumulative_delta
+        self.score_function = score_function
+        self.pass_function = pass_function
+        self.stop_function = stop_function
         self.trainer = trainer
         self.counter = 0
         self.best_score = None  # type: Optional[float]
-        self.logger = setup_logger(__name__ + "." + self.__class__.__name__)
+        self.min_delta = min_delta
+        self.cumulative_delta = cumulative_delta
 
     def __call__(self, engine: Engine) -> None:
         score = self.score_function(engine)
 
+        self._update_state(score)
```

Comment:
If we would have liked been even more general, the test would have been wrapped in a user function too. Anyway, that’s a detail.

Reply:
I was also considering that, but don't you think that it should be fixed since it's the No Improvement handler? 
The condition that there have been no improvements to the score in the last k epochs remains constant right otherwise it may become a little complicated.
Let me know what you think :)

Reply:
Let's do it like that :)